src/service.c

```diff
@@ -133,22 +147,60 @@ static gboolean r_on_handle_install(RInstaller *interface,
 	return r_on_handle_install_bundle(interface, invocation, arg_source, NULL);
 }
 
-static gboolean r_on_handle_info(RInstaller *interface,
+static GVariant* convert_bundle_info_to_dict(RaucManifest *manifest)
+{
+	GVariantDict dict;
+
+	g_variant_dict_init(&dict, NULL);
+
+	if (manifest->update_compatible)
+		g_variant_dict_insert(&dict, "compatible", "s", manifest->update_compatible);
+
+	if (manifest->update_version)
+		g_variant_dict_insert(&dict, "version", "s", manifest->update_version);
+
+	if (manifest->update_description)
+		g_variant_dict_insert(&dict, "description", "s", manifest->update_description);
+
+	if (manifest->update_build)
+		g_variant_dict_insert(&dict, "build", "s", manifest->update_build);
```

Comment:
Perhaps these should be part of a "update" dict to match the manifest structure?

Reply:
We can do this. First of all I thougth about aligning it with the current 'rauc info' json output which is

``` json
{
  "compatible" : "Test Config",
  "version" : "2011.03-2",
  "description" : null,
  "build" : null,
  "hooks" : [],
  "images" : [
    {
      "rootfs" : {
        "variant" : null,
        "filename" : "rootfs.img",
        "checksum" : "101a4fc5c369a5c89a51a61bcbacedc9016e9510e59a4383f739ef55521f678d",
        "size" : 8192,
        "hooks" : [],
        "adaptive" : []
      }
    },
    {
      "appfs" : {
        "variant" : null,
        "filename" : "appfs.img",
        "checksum" : "f95c0891937265df18ff962869b78e32148e7e97eab53fad7341536a24242450",
        "size" : 8192,
        "hooks" : [],
        "adaptive" : []
      }
    }
  ]
}
```

But since sections tend to increase probably just reflecting the actual manifest structure is more future-proof.

Reply:
In the example in https://github.com/rauc/rauc/pull/965#discussion_r983520569 I already addressed the "list of single value dicts" in "images", which we also should do at some point.

Reply:
Could only be fixed by breaking existing consumers I guess. Or by duplicating the info.

Reply:
We could also add a json-v2 format, which matches the layout of the InspectBundle method?